telegrinder/bot/rules/callback_data.py

```diff
@@ -46,7 +47,7 @@ async def check(self, event: CallbackQuery, ctx: dict) -> bool:
 
 
 class CallbackDataJsonModel(CallbackQueryRule):
-    def __init__(self, model: typing.Type[msgspec.Struct]):
+    def __init__(self, model: typing.Type[msgspec.Struct] | IsDataclass):
```

Comment:
```msgspec.json.Decoder``` parameter *type* expects the type and you hinted `IsDataclass`, correct to `type[IsDataclass]`.